solver/src/settlement_submission/flashbots_api.rs

```diff
@@ -99,4 +119,94 @@ impl FlashbotsApi {
         ensure!(status.is_success(), "status {}: {:?}", status, body);
         Ok(())
     }
+
+    /// Query gas_price for the current network state (simplest one for Flashbots)
+    pub async fn gas_price(&self) -> Result<EstimatedGasPrice> {
+        let body = serde_json::json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "eth_gasFees",
+            "params": [],
+        });
+        let response = self.client.post(URL).json(&body).send().await?;
+        let status = response.status();
+        let body = response.text().await?;
+        ensure!(status.is_success(), "status {}: {:?}", status, body);
+
+        match serde_json::from_str::<jsonrpc_core::Output>(&body) {
+            Ok(body) => match body {
+                jsonrpc_core::Output::Success(body) => {
+                    match serde_json::from_value::<FlashbotGasPrice>(body.result) {
+                        Ok(gas_price) => Ok(EstimatedGasPrice {
+                            eip1559: Some(GasPrice1559 {
+                                base_fee_per_gas: gas_price.base_fee_per_gas.to_f64_lossy(),
+                                max_fee_per_gas: gas_price.default.max_fee_per_gas.to_f64_lossy(),
+                                max_priority_fee_per_gas: gas_price
+                                    .default
+                                    .max_priority_fee_per_gas
+                                    .to_f64_lossy(),
+                            }),
+                            ..Default::default()
+                        }),
+                        Err(err) => Err(anyhow!("result not a FlashbotGasPrice: {}", err)),
+                    }
+                }
+                jsonrpc_core::Output::Failure(body) => Err(anyhow!(body.error)),
+            },
+            Err(err) => {
+                tracing::info!("flashbot cancellation response: {}", body);
+                Err(anyhow!(err))
+            }
+        }
```

Comment:
ubernit: I agree with @vkgnosis that we can still get rid of some "rightward drift" here. Maybe something like:
```suggestion
        let gas_price = serde_json::from_str::<jsonrpc_core::Output>(&body)
            .with_context(|err| {
                tracing::info!("flashbot cancellation response: {}", body);
                anyhow!("invalid Flashbots RPC response: {}", err)
            })
            .and_then(|output| match output {
                jsonrpc_core::Output::Success(body) => serde_json::from_value::<FlashbotGasPrice>(body.result)
                    .context("result not a FlashbotGasPrice"),
                jsonrpc_core::Output::Failure(body) => bail!("Flashbots RPC error: {}", body.error),
            })?;
        Ok(EstimatedGasPrice {
            eip1559: Some(GasPrice1559 {
                base_fee_per_gas: gas_price.base_fee_per_gas.to_f64_lossy(),
                max_fee_per_gas: gas_price.default.max_fee_per_gas.to_f64_lossy(),
                max_priority_fee_per_gas: gas_price
                    .default
                    .max_priority_fee_per_gas
                    .to_f64_lossy(),
            }),
            ..Default::default()
        })
```

Reply:
Yes, I still think the readability can be improved with early returns using `?` and `context` more. Also should convert the tracing log into context. I'm fine with merging either way but would be nice as a follow up.

Reply:
Oh ok, sorry, I thought the previous comment on readability was about that cumbersome `u64::from_str_radix` calls.
Anyway, this suggestion taken.